src/pages/Fox/components/FoxWifHat.tsx

```diff
@@ -1,25 +1,72 @@
-import { Box, Container, Heading, Image, useColorModeValue } from '@chakra-ui/react'
-import { foxAssetId } from '@shapeshiftoss/caip'
-import { useCallback } from 'react'
+import { Box, Container, Heading, Image, Skeleton, useColorModeValue } from '@chakra-ui/react'
+import type { AccountId } from '@shapeshiftoss/caip'
+import { foxWifHatAssetId, fromAccountId, fromAssetId } from '@shapeshiftoss/caip'
+import { bnOrZero } from '@shapeshiftoss/utils'
+import { useCallback, useMemo, useState } from 'react'
 import { useTranslate } from 'react-polyglot'
 import FoxWifHatIcon from 'assets/foxwifhat-logo.png'
 import { Text } from 'components/Text'
 import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { selectAccountIdsByChainId } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
 
+import { useFoxWifHatMerkleTreeQuery } from '../hooks/useFoxWifHatMerkleTreeQuery'
+import { FoxWifHatClaimModal } from './FoxWifHatClaimModal'
 import { FoxWifHatClaimRow } from './FoxWifHatClaimRow'
 
-// @TODO: replace with proper foxwifhat asset id
-const foxWifHatAssetId = foxAssetId
-
 export const FoxWifHat = () => {
   const translate = useTranslate()
   const isFoxWifHatEnabled = useFeatureFlag('FoxPageFoxWifHatSection')
   const containerBackground = useColorModeValue('blackAlpha.50', 'whiteAlpha.50')
+  const accountIdsByChainId = useAppSelector(selectAccountIdsByChainId)
+  const [isClaimModalOpened, setIsClaimModalOpened] = useState(false)
+  const [claimAccountId, setClaimAccountId] = useState<AccountId | undefined>()
+  const getFoxWifHatClaimsQuery = useFoxWifHatMerkleTreeQuery()
+
+  const handleClaimModalClose = useCallback(() => {
+    setClaimAccountId(undefined)
+    setIsClaimModalOpened(false)
+  }, [setClaimAccountId, setIsClaimModalOpened])
+
+  const handleClaimModalOpen = useCallback(
+    (accountId: AccountId) => {
+      setClaimAccountId(accountId)
+      setIsClaimModalOpened(true)
+    },
+    [setClaimAccountId, setIsClaimModalOpened],
+  )
+
+  const claimRows = useMemo(() => {
+    const accountIds = accountIdsByChainId[fromAssetId(foxWifHatAssetId).chainId]
 
-  const handleClaim = useCallback(() => {
-    // eslint-disable-next-line no-console
-    console.log('Claim')
-  }, [])
+    if (getFoxWifHatClaimsQuery.isLoading || !getFoxWifHatClaimsQuery.data)
+      return <Skeleton height='64px' width='100%' />
+
+    return Object.entries(getFoxWifHatClaimsQuery.data.claims).map(([address, claim]) => {
+      const accountId = accountIds?.find(
+        accountId => fromAccountId(accountId).account === address.toLowerCase(),
+      )
+
+      if (!accountId) return null
+
+      return (
+        <FoxWifHatClaimRow
+          key={accountId}
+          accountId={accountId}
+          amountCryptoBaseUnit={bnOrZero(claim.amount).toFixed()}
```

Comment:
blocking: this is coming from the wire as hex:

<img width="414" alt="image" src="https://github.com/user-attachments/assets/74609f02-a5b7-4da9-97db-3c0e49cdb5d8" />

And should be converted to a stringified bn first, ideally with a react-query `select` fn

Reply:
Potentially related to issue re: missing FWH asset however, but still prefer to un-hex this guy for sanity purposes 🙏🏽

Reply:
Done in [7e9a198](https://github.com/shapeshift/web/pull/8611/commits/7e9a198a552c9ffbe24675fdb83a91d1875e342b)